packages/services/api/src/modules/schema/providers/schema-version-helper.ts

```diff
@@ -302,4 +311,34 @@ export class SchemaVersionHelper {
   async getIsValid(schemaVersion: SchemaVersion) {
     return schemaVersion.isComposable && schemaVersion.hasContractCompositionErrors === false;
   }
+
+  /**
+   * There's a possibility that the composite schema SDL contains parts of the supergraph spec.
+   * This is a problem because we want to show the public schema to the user, and the supergraph spec is not part of that.
+   * This may happen when composite schema was produced with an old version of `transformSupergraphToPublicSchema`
+   * or when supergraph sdl contained something new.
+   *
+   * This function will check if the SDL contains supergraph spec and if it does, it will transform it to public schema.
+   */
+  private autoFixCompositeSchemaSdl(sdl: string, versionId: string) {
+    if (containsSupergraphSpec(sdl)) {
+      this.logger.warn(
+        'Composite schema SDL contains supergraph spec, transforming to public schema (versionId: %s)',
+        versionId,
+      );
+      const transformedSdl = print(
+        transformSupergraphToPublicSchema(parseGraphQLSource(sdl, 'autoFixCompositeSchemaSdl')),
+      );
+
+      this.logger.debug(
+        transformedSdl === sdl
+          ? 'Transformation did not change the original SDL'
+          : 'Transformation changed the original SDL',
+      );
+
+      return transformedSdl;
+    }
+
+    return sdl;
```

Comment:
this should be outside the if block? We can also flip the condition for early return and avoid indent